### PR TITLE
FIX: key ESC/RETURN handled in Options Dialog & KasPathEdit

### PR DESCRIPTION
**Pull request by rich2014**
fix key ESC/RETURN handled in KasPathEdit & Options Dialog:

1. after the Key ESC/RETURN handled in TKASPathEdit.handleSpecialKeys(), Key should be eaten (there is a bug in #876)

2. in fOptions, set `btnCancel.Cancel`=`true` , and remove `alOptionsActionList`, handle `ESCAPE` correctly on all platforms

3. tested on Windows 11, MacOS12 and Linux

### CODE REVIEW

File: src/foptions.pas [hunk 1]
```diff
@@ -222,7 +219,7 @@ procedure TfrmOptions.FormClose(Sender: TObject; var CloseAction: TCloseAction);
 
 procedure TfrmOptions.FormCloseQuery(Sender: TObject; var CanClose: boolean);
 begin
-  CanClose := (ModalResult in [mrOK, mrCancel]) or CycleThroughOptionEditors(False);
```
Comment (alexx2000) [inline src/foptions.pas:225]: Why do you remove `mrCancel`? DC should not ask user to save changes when user press cancel. It should ask only when user close window by "X" title button.

Comment (rich2014) [reply src/foptions.pas:225]: for careless closing the Form.

press Cancel with quering, it seems a bit risky.

Comment (alexx2000) [reply src/foptions.pas:225]: It was such way five years ago. It is very annoying, I press `Cancel` when I want to cancel all changes. Why DC should ask me again about this?

Comment (avmaksimov) [reply src/foptions.pas:225]: > It was such way five years ago. It is very annoying, I press `Cancel` when I want to cancel all changes. Why DC should ask me again about this?

I think it's normal situation if DC will asks about it too. You could by mistake press another button (.

**Comment by rich2014** [comment]
@alexx2000

**Pull request merged**